### ISSUE TITLE
bug(MatChipList): chip list disabled should respect each chip item disabled state

### ISSUE THREAD

**Issue by Nikami** [angular/components#27971]
### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### Description

Chiplist [disabled] toggling from disable to enable doesn't respect each chip item [disabled] state. Chip items, which are actually disabled become visually enabled.

### Reproduction

StackBlitz link: https://stackblitz.com/edit/vdefsz?file=src%2Fexample%2Fchips-input-example.html,src%2Fexample%2Fchips-input-example.ts

Steps to reproduce:
1. The chip item is disabled and has properly disabled styles in initialization
![2023-10-20_15h12_54](https://github.com/angular/components/assets/12629525/a8ca53f3-a9ee-44a7-8e4b-f8f20f74b469)
2. Click 'Disable chiplist'
3. Click 'Enable chiplist'

### Expected Behavior

The chip item is visually disabled 

### Actual Behavior

The chip item ignores its disabled state and looks like enabled

### Environment

- Angular: 16.2.9
- CDK/Material: 16.2.9
- Browser(s): Google Chrome, Version 118.0.5993.88 (Official Build) (64-bit)
- Operating System (e.g. Windows, macOS, Ubuntu): Windows 10

**Comment by amysorto** [comment]
I am unable to reproduce this issue on the linked Stackblitz. Can you take another look at it just to make sure you still see the issue there?

**Comment by Nikami** [comment]
> I am unable to reproduce this issue on the linked Stackblitz. Can you take another look at it just to make sure you still see the issue there?

Yes, still reproduces with the same behavior. Chip **item** starts to be enabled, while it's originally disabled. Chiplist is ignoring **isDisabled: true** property on its chips after toggling the disable/enable state. 


![bug1](https://github.com/angular/components/assets/12629525/4e455363-1d83-434a-a068-69c9f444fb16)

**Comment by mmalerba** [comment]
Yes this seems like a real issue to me, I see the same behavior
